wrappers/rust/icicle-curves/icicle-bw6-761/src/curve.rs

```diff
@@ -1,30 +1,57 @@
+#[cfg(all(feature = "arkworks", feature = "g2"))]
+use ark_bw6_761::g2::Config as ArkG2Config;
 #[cfg(feature = "arkworks")]
-use ark_bw6_761::{g1::Config as ArkG1Config, Fq, Fr};
+use ark_bw6_761::{g1::Config as ArkG1Config, Fq};
+use icicle_bls12_377::curve::BaseField as bls12_377BaseField;
 use icicle_core::curve::{Affine, Curve, Projective};
-use icicle_core::field::{Field, MontgomeryConvertibleField};
-use icicle_core::traits::{FieldConfig, FieldImpl, GenerateRandom};
-use icicle_core::{impl_curve, impl_field, impl_scalar_field};
+use icicle_core::field::Field;
+use icicle_core::traits::FieldConfig;
+use icicle_core::{impl_curve, impl_field};
 use icicle_cuda_runtime::device_context::{get_default_device_context, DeviceContext};
 use icicle_cuda_runtime::error::CudaError;
 use icicle_cuda_runtime::memory::HostOrDeviceSlice;
 
-pub(crate) const SCALAR_LIMBS: usize = 6;
 pub(crate) const BASE_LIMBS: usize = 12;
 
-impl_scalar_field!("bw6_761", SCALAR_LIMBS, ScalarField, ScalarCfg, Fr);
 impl_field!(BASE_LIMBS, BaseField, BaseCfg, Fq);
-impl_curve!("bw6_761", CurveCfg, ScalarField, BaseField);
+pub type ScalarField = bls12_377BaseField;
+impl_curve!(
+    "bw6_761",
+    bw6_761,
+    CurveCfg,
+    ScalarField,
+    BaseField,
+    ArkG1Config,
```

Comment:
What if arkworks feature is not enabled?

Reply:
Still compiles because `ArkG1Config` and other similar names are treated as meaningless unused idents if the feature is disabled